src/features/top-level-await.md

```diff
@@ -71,6 +71,18 @@ try {
 }
 ```
 
+### Polyfills
+
+You can effectively hold the execution of a static import by `await`-ing a conditional, dynamic import. This is very useful when using polyfills that you only want to load when the browser needs them:
+
+```js
+if (browserNeedsPolyfill()) {
+  await import("./polyfill.mjs");
+}
+
+import "./main.js";
```

Comment:
Maybe include an example of multiple polyfills as well (to show that you don't want to import them sequentially but rather with `await Promise.all([...])`).

Reply:
Oh! You could use `Array#reduce` to build the array of needed polyfills! `/me runs`

Reply:
Does this actually work? My thought was that sibling modules could execute before the async import?

Reply:
Siblings can indeed execute before if you have an await. Also if this is in the "Root Node" of an application it won't even execute until the entire graph is done.

Further... it would appear that according to the [HTML integration](https://github.com/whatwg/html/pull/4352) that one script tag would not block a second script tag (follow similar sibling behavior)... so honestly I'm unsure that `await` is a reliable / good mechanism for polyfills

/cc @littledan and @domenic who might have other opinions here.

Reply:
That definitely makes this pattern unviable then, which I was unsure about and was the whole reason I looped you in in the first place :D Ignore this PR then.

Reply:
One more:

Let's call this module `entrypoint.js`, and it has this TLA and a later `import "main.js"` (or a specifier import/default import, etc). Can `main.js` be executed before the TLA resolves?

Definitely `entrypoint.js`'s siblings could execute while the TLA is resolving.

Reply:
Huh, I'm pretty confused by this discussion. I think it's good to show people how they can conditionally load polyfills, so that the practice of unconditionally including all the polyfills can be reduced. I always thought that this was one of the main motivating use cases for top-level await.

Don't you want to encourage people to include the polyfills at the application integration point, rather than deeper in a module graph? If so, I think this pattern works.

I'm confused the discussion of multiple script tags. This example doesn't use multiple script tags, but `<script type=module>` is already not a load-blocking script, and yet continues to happen in order per HTML semantics. This is just like if your script tag had a fetch and .then to eval it, to load a polyfill--no better or worse.

Reply:
> Don't you want to encourage people to include the polyfills at the application integration point, rather than deeper in a module graph? If so, I think this pattern works.

As a practice, yes. But I don't think TLA actually works this way. What this code wants is for `main.js` to load _after_ `polyfill.js`. But I don't think we prevent `main.js` from executing while we wait for `polyfill.js` to load.

But maybe I'm wrong, so that's why I asked. Maybe we can depend on `polyfill.js` executing before `main.js`.

> I'm confused the discussion of multiple script tags.

I think that came up because it's hard to distinguish between which sibling we're talking about. Definitely siblings of the module that includes a TLA can execute first. But can modules requested (eg, `main.js`) by the module that includes a TLA (this code example, or `entrypoint.js`) execute before the TLA (`polyfill.js`)?

Reply:
> But I don't think we prevent main.js from executing while we wait for polyfill.js to load.

Oh, that's right, it would have to be a dynamic import of "main.js", not a static import. Static imports are all hoisted. This doesn't relate to sibling execution.